devlib/trace/perf.py

```diff
@@ -16,34 +16,48 @@
 
 import os
 import re
+import time
 from past.builtins import basestring, zip
 
 from devlib.host import PACKAGE_BIN_DIRECTORY
 from devlib.trace import TraceCollector
 from devlib.utils.misc import ensure_file_directory_exists as _f
 
 
-PERF_COMMAND_TEMPLATE = '{} stat {} {} sleep 1000 > {} 2>&1 '
+PERF_COMMAND_TEMPLATE = '{binary} {command} {options} {events} sleep 1000 > {outfile} 2>&1 '
+PERF_REPORT_COMMAND_TEMPLATE= '{binary} report {options} -i {datafile} > {outfile} 2>&1 '
+PERF_RECORD_COMMAND_TEMPLATE= '{binary} record {options} {events} -o {outfile}' 
 
-PERF_COUNT_REGEX = re.compile(r'^(CPU\d+)?\s*(\d+)\s*(.*?)\s*(\[\s*\d+\.\d+%\s*\])?\s*$')
-
-DEFAULT_EVENTS = [
+PERF_DEFAULT_EVENTS = [
     'migrations',
     'cs',
 ]
 
+SIMPLEPERF_DEFAULT_EVENTS = [
+    'raw-cpu-cycles',
+    'raw-l1-dcache',
+    'raw-l1-dcache-refill',
+    'raw-instructions-retired',
```

Comment:
`raw-instructions-retired` is an invalid event on my device (nexus5) 
Are there a set of predefined values that should be available on all devices?

Reply:
Yes that was a last minute typo sorry, should have been instruction instead of instructions. I have updated